### DISCUSSION TITLE
Is there a way to have a dinamically created name value?

### DISCUSSION THREAD

**Discussion by andresdellachiesa** [react-hook-form/react-hook-form#2558] [Q&A]
I have a select tag which name is attached to an state hook. Basically, I want the name of the select to change dinamically, depending on the selected option value. 

It goes as it follows:

```
    const [right, setRight] = useState("NameOne")

    <select name={right} ref={register} onChange={e => handleRights(e.target.value)}>
        <option value="nameOne">Name One</option>
        <option value="NameTwo">Name Two</option>
    </select>
```

I use the handleRights handler to set the hook state:

```
    const handleRights = (value) => {
        setRight(value);
    }
```

Nevertheless, value is always "NameOne". That means: the initial state. 

Is there a way to achieve this behavior?

Thanks in advance.

**Comment by bluebill1049** [comment]
you are mixed controlled with uncontrolled, take a look `watch` API
